op-node/rollup/derive/data_source.go

```diff
@@ -93,20 +93,25 @@ type DataSourceConfig struct {
 // isValidBatchTx returns true if:
 //  1. the transaction has a To() address that matches the batch inbox address, and
 //  2. the transaction has a valid signature from the batcher address
-func isValidBatchTx(tx *types.Transaction, l1Signer types.Signer, batchInboxAddr, batcherAddr common.Address, logger log.Logger) bool {
+func isValidBatchTx(g eth.GenericTx, l1Signer types.Signer, batchInboxAddr, batcherAddr common.Address, logger log.Logger) (tx *types.Transaction, isValid bool) {
+	tx, err := g.Transaction()
```

Comment:
We need to be stricter here, and only allow the allowed batcher types (see specs PR https://github.com/ethereum-optimism/specs/pull/518)